### DISCUSSION TITLE
What can be done to improve the Deepin Store

### DISCUSSION THREAD

**Discussion by serial02** [linuxdeepin/developer-center#3216] [Features Request & Ideas | 特性请求 & 头脑风暴]
Hello everybody!

 Everyone is technical and crazy, everyone has a little and formed opinion about everything, where some are good, others not so much, many based on deep knowledge, others on guesswork, others by mere predilection of something without measuring consequences and, many opinions even meaningless. for lack of knowledge.

 So, here are my opinions, impressions or suggestions about the Deepin Distro that I've been using since version 15.11, and that I believe deserves a place in the Sun of Prominence. 

 Lets go!

 I still believe the Deepin Store needs to change, and fast:

It has improved a lot since the release of the V20 version, but it needs to have the updates (LATEST) available for the vast majority of applications available in the Linux world, not enough for the Deepin Store to only have the largest number of packages. 

 Why do I say this?

A clear example is the range of outdated browsers on the Deepin Store, what for?  To force the user to use Uniontech's Deepin Browser?  Well, those who like Distro Deepin are not always willing to use the standard that you “impose” and, if the user cannot use a browser of their personal taste, updated with the latest security fixes, as well as their extensions, whatever it is, or an email client of your choice, you will for sure stop using the distro and switch to another that allows this possibility.  THIS IS A FACT!

The options you've developed may even be safe, beautiful and everything else, but the vast majority of users worldwide speaking are not aware of them and, until you gain confidence (credibility for your daily use), it takes time!  Therefore, the most popular options in the browser market need to be available in the store and there has already been a big step in this direction: - Google-Chrome, Firefox, Vivaldi (updated OK with the latest versions).  Opera, Brave (outdated) and Microsotf-Edge-Stable (Updated), but I didn't find it in the store, only in the terminal which is abbreviated to softedge, in fact a great difficulty in finding applications through the terminal using search.  As for the Thor browser, I didn't find it through the terminal or the Deepin Store.  Many people migrate from Windows to Linux and do not have in-depth knowledge.  They want to install the product, their favorite packages and use it without any problems looking for the correct language and etc... Make life easier for the BEGINNER user!

There are thousands of repositories that can be reviewed, their applications tested and made available by the team that develops the Deepin Store, instead of the common user having to resort to the official repository of browsers, compromising any type of application use in the Deepin Distro.

 A lot of things we install out of the box from the Deepin Store give errors or don't work!  This is a fact that needs to be taken into account, as many dependencies are old and do not correspond to new versions of applications or the Distro in question.  Analysis is necessary first of all, of course!  An example of this is here in a flatpak version of Brave Browser already reported on the Deepin control center on Git ( https://github.com/linuxdeepin/developer-center/issues/3211 ).

I can't understand why some browsers are NOT updated in the Deepin store, like Opera Browser which is still in version 76. How many flaws and vulnerabilities have not been fixed in Opera since version 76?  Why insist on keeping the browser out of date in the Store?

Another example is LibreOffice! 

 For each version made available by the Official LibreOffice provider, around 100 security vulnerabilities or more are fixed!  So I ask?  Why use an outdated version of any application at all?  For me, this is to belittle the work of the Engineering Team that develops the LibreOffice application for Global use and to put at risk the users who give credibility to the use of the application by any Distro of their choice.

 Of course, these updates cannot happen right away and take some time, but after a test period and, after checking the circumstances and, if everything is ok, they will be made available to users who are fans of the Distro as soon as possible.  See the example of Arch, Fedora, Manjaro among others.

In the DistroWatch ranking ( https://distrowatch.com/ ) of the top 10 distros placed, three use the latest applications and kernels, updated daily and constantly, being EndeavorsOS, Fedora and Manjaro.  

 Still on the Deepin Store, I believe that developers need to give more package options, snap, .deb, flatpak and even the possibility in the near future, the rpm-ostree example, as well as this other example of ZorinOS, where opening the ZorinOS store, we are faced with all these options as shown in the image below.  In my understanding, positive point for this ZorinOS Distro.

 Please, the purpose of everything above is just for reflection to open the range to possibilities

![image](https://user-images.githubusercontent.com/32621790/171641175-a34d9cd8-cd68-4982-8f83-55a41aae1df8.png)

**Comment by serial02** [comment]
Below are examples of applications not found in the Deepin Store, but appear via the terminal, where it is outdated.

![Capturar_select-area_20220602111706](https://user-images.githubusercontent.com/32621790/171650673-e994357e-ad30-42ed-92e9-f3da817560b5.png)

![Capturar_dde-desktop_20220602092325](https://user-images.githubusercontent.com/32621790/171650805-5f3c9abe-0340-4815-88e8-b6a9c3b28f17.png)

**Comment by raspin0** [comment]
Yeah, maintenance of deepin repositories leave much to be desired. Dependency wise as version wise. However, there are no conspiracy theories about enforcing their solutions lol.  

Honestly, supporting all packaging formats on deepin will be problematic because:
Snap:   
- Will slow booting up times
- doesn't adapt cursor theme
- doesn't apply native QT/GTK theming
- the time to launch an app for the first time is unacceptable
- it's not stable enough for production environment
- doesn't elevate native dialogs
- plenty of issues with some system specific solutions like hardware acceleration

Flatpak:
- Installing anything downloads not only the application, but also it's running environment
- doesn't autoremove old dependencies

Snap+Flatpak: 
- The biggest fundamental flaw of both is space management, while deepin already isn't the greatest with a/b root + separate backup
- deepin doesn't elevate portals afaik., which means that applications wont use native file dialogs etc.
- **both have less performance than native packaging**

Ultimately I think that deepin is here to unify the experience. It's not good to have so many solutions at once. Every packaging format out there has some fundamental unacceptable flaws and I'm waiting with eagerness to see how deepin developers will tackle the problem. 

_Honestly, I wonder why gnu/linux has such a terrible package management. On windows it may be less secure, but it's simple. On android you have the play store and eventually you can sideload .apk file or use third party store(which uses the same technology). On linux however nothing can be straightforward. You can't install a snap app using a file, you can with flatpak but most distributions don't support it. You don't have one store and must search and use every solution to have an access to 100% linux software. Yeah, the decentralized model is definitely way to go..._

**Comment by BLumia** [comment]
Since the deepin distro itself is based on a pretty old version (currently old stable) of debian, so packages from the deepin store are updated by people from the deepin team manually. Most packages are maintained by the people from deepin instead of community maintainer so it means it's most likely a less-mainstream package will be outdated. There is a button to flag outdated package but still it requires people from the deepin team to update the package manually.

In the long run, we are planning to introduce ways to let the community maintain packages that are not in the official repo, although details are still not decided yet, using PPA and using things like snap and flatpak is also possible for now.

The deepin store app currently will only list managed source of packages since it's actually mainly for enterprise usage and isn't a cross-distro solution, we're not yet sure if we can introduce a better integration of snap or flatpak. But there are still some 3rd-party store implementation that might worth to check it out, like KDE's [Discovery](https://apps.kde.org/discover/), and [Spark App Store](http://spark-app.store/).

Hope that helps :)
